From 915ccfae06d3bc370790b4e2120c8e76694b7507 Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@foundries.io>
Date: Fri, 29 Jan 2021 22:37:35 +0200
Subject: [PATCH 1/5] [FIO internal] imx8mp-lpddr4-evk: fix build issues when
 SPL_DM=y

1. Force disable DM_I2C for the SPL build (workaround)
2. Disable usage of TCPC from the board file for the SPL build, as
from board/freescale/common/Makefile:

ifndef CONFIG_SPL_BUILD
obj-$(CONFIG_USB_TCPC) += tcpc.o
endif

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
Signed-off-by: Tim Anderson <tim.anderson@foundries.io>
---
 board/freescale/imx8mp_evk/imx8mp_evk.c | 12 ++++++------
 include/configs/imx8mp_evk.h            |  1 +
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/board/freescale/imx8mp_evk/imx8mp_evk.c b/board/freescale/imx8mp_evk/imx8mp_evk.c
index ff46a2b252..aa0f11c488 100644
--- a/board/freescale/imx8mp_evk/imx8mp_evk.c
+++ b/board/freescale/imx8mp_evk/imx8mp_evk.c
@@ -182,7 +182,7 @@ int board_phy_config(struct phy_device *phydev)
 }
 #endif
 
-#ifdef CONFIG_USB_TCPC
+#if defined(CONFIG_USB_TCPC) && !defined(CONFIG_SPL_BUILD)
 struct tcpc_port port1;
 struct tcpc_port port2;
 
@@ -412,7 +412,7 @@ int board_usb_init(int index, enum usb_init_type init)
 	imx8m_usb_power(index, true);
 
 	if (index == 0 && init == USB_INIT_DEVICE) {
-#ifdef CONFIG_USB_TCPC
+#if defined(CONFIG_USB_TCPC) && !defined(CONFIG_SPL_BUILD)
 		ret = tcpc_setup_ufp_mode(&port1);
 		if (ret)
 			return ret;
@@ -420,7 +420,7 @@ int board_usb_init(int index, enum usb_init_type init)
 		dwc3_nxp_usb_phy_init(&dwc3_device_data);
 		return dwc3_uboot_init(&dwc3_device_data);
 	} else if (index == 0 && init == USB_INIT_HOST) {
-#ifdef CONFIG_USB_TCPC
+#if defined(CONFIG_USB_TCPC) && !defined(CONFIG_SPL_BUILD)
 		ret = tcpc_setup_dfp_mode(&port1);
 #endif
 		return ret;
@@ -439,7 +439,7 @@ int board_usb_cleanup(int index, enum usb_init_type init)
 	if (index == 0 && init == USB_INIT_DEVICE) {
 		dwc3_uboot_exit(index);
 	} else if (index == 0 && init == USB_INIT_HOST) {
-#ifdef CONFIG_USB_TCPC
+#if defined(CONFIG_USB_TCPC) && !defined(CONFIG_SPL_BUILD)
 		ret = tcpc_disable_src_vbus(&port1);
 #endif
 	} else if (index == 1 && init == USB_INIT_HOST) {
@@ -452,7 +452,7 @@ int board_usb_cleanup(int index, enum usb_init_type init)
 	return ret;
 }
 
-#ifdef CONFIG_USB_TCPC
+#if defined(CONFIG_USB_TCPC) && !defined(CONFIG_SPL_BUILD)
 /* Not used so far */
 int board_typec_get_mode(int index)
 {
@@ -484,7 +484,7 @@ int board_typec_get_mode(int index)
 
 int board_init(void)
 {
-#ifdef CONFIG_USB_TCPC
+#if defined(CONFIG_USB_TCPC) && !defined(CONFIG_SPL_BUILD)
 	setup_typec();
 #endif
 
diff --git a/include/configs/imx8mp_evk.h b/include/configs/imx8mp_evk.h
index a89d161122..d852f3f662 100644
--- a/include/configs/imx8mp_evk.h
+++ b/include/configs/imx8mp_evk.h
@@ -33,6 +33,7 @@
 #define CONFIG_POWER
 #define CONFIG_POWER_I2C
 #define CONFIG_POWER_PCA9450
+#undef CONFIG_DM_I2C
 
 #define CONFIG_SYS_I2C
 
-- 
2.17.1

